qiskit/providers/aer/openpulse/qobj/digest.py

```diff
@@ -0,0 +1,575 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=eval-used, exec-used, invalid-name
+
+"""A module of routines for digesting a PULSE qobj
+into something we can actually use.
+"""
+
+from collections import OrderedDict
+import numpy as np
+import numpy.linalg as la
+from .op_system import OPSystem
+from .opparse import HamiltonianParser, NoiseParser
+from .operators import qubit_occ_oper_dressed
+from ..solver.options import OPoptions
+# pylint: disable=no-name-in-module,import-error
+from ..cy.utils import oplist_to_array
+from . import op_qobj as op
+
+
+def digest_pulse_obj(qobj):
+    """Takes an input PULSE obj an disgests it into
+    things we can actually make use of.
+
+    Args:
+        qobj (Qobj): Qobj of PULSE type.
```

Comment:
this function actually takes a raw dict, not a Qobj object.
It should be changed to take Qobj because that simplifies the function. Currently it is doing all these `dict.get()` stuff, but the whole point of a validated Qobj is that you can assume all those fields exist, or at least it will return None to you if you do `qobj.foo`.